roles/pool/src/lib/mining_pool/mod.rs

```diff
@@ -200,11 +200,11 @@ impl Downstream {
     }
 
     pub async fn next(self_mutex: Arc<Mutex<Self>>, mut incoming: StdFrame) -> PoolResult<()> {
-        let message_type = incoming
-            .get_header()
-            .ok_or_else(|| PoolError::Custom(String::from("No header set")))?
-            .msg_type();
-        let payload = incoming.payload();
+        let message_type = incoming.header().msg_type();
+        let payload = match incoming.payload() {
+            Some(p) => p,
+            None => return Err(PoolError::Custom(String::from("No payload set"))),
```

Comment:
just out of curiosity. Why here you use String::from and in other places you use .to_string() . From what I concern you can just resolve this conversation and ignore my comment.

Reply:
Also here I would just have use expect since if there is an issue we are defenetly in an unrecoverable situation. But more then ok then merging it like that

Reply:
tbh everywhere outside the `framing` crate I was just following whatever is already their with minimum effort to improve/change it because otherwise those prs become to big to review